### PR TITLE
[1] Store and Products Screens styling

### PR DESCRIPTION
**Pull request by tommypoa**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
### Notable changes to BaseComponents.js
- Allow `ButtonLabel` to take color props instead of filledButton
- Allow `FilledButtonContainer` to take in `width` and `height` props

### Styling!
- A ton of containers added to style sheets in `styled` folder
   - A couple of containers in `styled/shared.js` file that might be reused in future styling, in effort to reduce in-line styling
   - `EBTStatusBar` and `SearchBar` implemented in this folder rather than as a component - they seem like specialized Views/Buttons that only require styling and no use for re-usability
   - `FlatList` styles in `styled.store.js`
- `ProductDetailsScreen` styled with additional containers 
- `StoreListScreen`
   - Search bar and color
   - 'See Products' powered by `StoreProductButton.js` (see **New Components**)
- `BottomSheet`
   - `ProductCard` sized
   - Deleted categories, centralized as 'Products'
### New Components
- `StoreProductButton` (might need a naming fix?)

### Deletion / Replacements
- Deleted `SpecialNotice.js` and moved notice to `ProductInfo.js`

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### Online sources

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"
- Styled Components with ReactNative: https://dev.to/acro5piano/use-styled-components-reactnative-s-flatlist-in-typescript-308e
- Useful Flexbox guide: https://css-tricks.com/snippets/css/a-guide-to-flexbox/
### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
- #33: BaseComponents for Store / Products. Current branch is branched of that of #33.
## How to review

[//]: # "The order in which to review files and what to expect when testing locally"
- Made a ton of (quick) decisions (on creating new Containers, creating shared styled components) that may or may not be the best way forward. Please poke at it!
- **Bug finding**: Look and touch all components in `MapScreen`, `StoreListScreen`, `ProductDetailsScreen` and try catching visual/styling bugs
- **Code quality**: Have a mental map of 2 groups -- the stylesheets and the screens/components. Look out especially for redundant stylesheet additions + any funky JavaScript implementations in the screens, **and suggest an alternative if possible!**
## Next steps

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"
- Edge cases in `StoreListScreen`
   - Long store names
- Creating a separate stylesheet to house all styles for React Native elements/components (i.e FlatList, MapView)
- Restricting BottomSheet from moving upwards to expose the bare white screen at the bottom of the map.
- Replace top / bottom styling for `ButtonLabel` - there should be a better way to center label vertically.
- Importing Poppins for search bar in `StoreListScreen`
- Drop shadow for `BottomSheet`
## Tests Performed, Edge Cases

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"
- All behaviour on MapScreen.js and StoreListScreen.js working as before.

### Screenshots

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"
- ProductDetails
image
- BottomSheet
image
- Map search bar
image
- StoreListScreen (with edge cases)
image

CC: @wangannie @12aschen @kennethlien 

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: components/store/StoreCard.js [hunk 1]
```diff
@@ -1,41 +1,56 @@
...
+        <MaterialCommunityIcons name="directions" size={16} />
+        <Body color={Colors.secondaryText}>{address}</Body>
+      </InLineContainer>
+      <InLineContainer>
+        <FontAwesome name="clock-o" size={12} />
+        <Body color={Colors.secondaryText}>{hours}</Body>
+      </InLineContainer>
+
+      {/* Temporary EBT status; TODO @tommypoa to replace with tag */}
+      <Body color={Colors.secondaryText}>
```
Comment (anniero98) [inline components/store/StoreCard.js]: This should be removed now, right?

File: components/product/StoreProducts.js [hunk 2]
```diff
@@ -1,40 +1,28 @@
...
-import { TextButton } from '../BaseComponents';
+import {
+  ButtonContainer,
+  ButtonLabel,
+  Title,
+  Subhead
+} from '../BaseComponents';
 import StoreCard from '../store/StoreCard';
-import Product from './Product';
-
```
Comment (kennethlien) [inline components/product/StoreProducts.js]: Thank you for this filter idea, saved my booty

File: components/store/StoreCard.js (hunk 1 above)
Comment (kennethlien) [reply components/store/StoreCard.js]: Agreement

File: components/BaseComponents.js [hunk 3]
```diff
@@ -23,20 +23,20 @@ export const ButtonLabel = styled(TextButton)`
   display: flex;
   align-items: center;
   text-align: center;
-  letter-spacing: 0.01;
+  letter-spacing: 0.01px;
   text-transform: uppercase;
-  color: ${props => (props.filledButton ? Colors.white : Colors.black)};
+  color: ${props => (props.color ? props.color : Colors.black)};
```
Comment (emilyzhong) [inline components/BaseComponents.js]: nit: `${props => props.color || Colors.black}`

File: components/BaseComponents.js [hunk 4]
```diff
@@ -23,20 +23,20 @@ export const ButtonLabel = styled(TextButton)`
...
 `;
 
 export const FilledButtonContainer = styled(ButtonContainer)`
-  width: 344px;
-  height: 40px;
+  width: ${props => (props.width ? props.width : '344px')};
+  height: ${props => (props.height ? props.height : '40px')};
```
Comment (emilyzhong) [inline components/BaseComponents.js]: Similar nit as above

File: components/store/StoreCard.js [hunk 5]
```diff
@@ -1,41 +1,56 @@
...
-            <Body color={Colors.secondaryText}>Yes</Body>
-          ) : (
-            <Body color={Colors.secondaryText}>No</Body>
+    <StoreCardContainer>
+      <SpaceBetweenRowContainer>
+        <SpaceAroundRowContainer>
+          <Title color={Colors.activeText}>{name}</Title>
+          {ebt && (
+            <EBTStatusBar>
+              <FontAwesome name="check" size={12} color="#004E14" />
```
Comment (emilyzhong) [inline components/store/StoreCard.js]: Use `Colors.darkerGreen`

File: components/store/StoreCard.js [hunk 6]
```diff
@@ -1,41 +1,56 @@
...
+        </SpaceAroundRowContainer>
+        {seeProduct && <StoreProductButton callBack={callBack} />}
+      </SpaceBetweenRowContainer>
+      <Caption color={Colors.secondaryText}>Distance: {distance} mi </Caption>
+      <InLineContainer>
+        <MaterialCommunityIcons name="directions" size={16} />
+        <Body color={Colors.secondaryText}>{address}</Body>
+      </InLineContainer>
+      <InLineContainer>
+        <FontAwesome name="clock-o" size={12} />
```
Comment (emilyzhong) [inline components/store/StoreCard.js]: @ `MaterialCommunityIcon` and `FontAwesome` — probably best to use the same font library so you can be consistent with style and size.

File: components/store/StoreProductButton.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+function StoreProductButton({ callBack }) {
+  //   const { name, hours, address, distance, ebt } = store;
+  return (
+    <FilledButtonContainer
+      width="130px"
+      height="35px"
+      color={Colors.primaryOrange}
+      onPress={callBack}>
+      <ButtonLabel color="#fff">See Products</ButtonLabel>
```
Comment (emilyzhong) [inline components/store/StoreProductButton.js]: use `Colors.lightest`

File: screens/map/MapScreen.js [hunk 8]
```diff
@@ -16,13 +17,14 @@ import BottomSheet from 'reanimated-bottom-sheet';
...
-import { Subtitle } from '../../styled/shared';
+import { Body, ButtonLabel } from '../../components/BaseComponents';
 import {
   SearchBar,
-  StoreModal,
-  StoreModalBar,
-  TopText
+  BottomSheetContainer,
+  BottomSheetHeaderContainer,
+  StoreModalBar
```
Comment (emilyzhong) [inline screens/map/MapScreen.js]: this is a nit but try to use hanging commas so future diffs won't include extra lines

File: screens/map/MapScreen.js [hunk 9]
```diff
@@ -206,7 +196,8 @@ export default class MapScreen extends React.Component {
                 navigation: this.props.navigation
               })
             }>
-            <Subtitle>Search</Subtitle>
+            <FontAwesome name="search" size={12} style={{ color: '#f07723' }} />
```
Comment (emilyzhong) [inline screens/map/MapScreen.js]: `Colors.primaryOrange`

File: styled/product.js [hunk 10]
```diff
@@ -1,27 +1,30 @@
...
 
-export const ProductContainer = styled.View`
-  background-color: #fff;
-  border-radius: 10px;
-  padding: 20px;
-  margin: 5% 5%;
-  box-shadow: 0px 10px 20px #e3e1e1;
+export const ProductInfoContainer = styled.View`
+  width: 80%;
+  padding: 5%;
```
Comment (emilyzhong) [inline styled/product.js]: Generally padding is done in pixels!

File: styled/product.js [hunk 11]
```diff
@@ -1,27 +1,30 @@
...
+export const ProductInfoCaptionContainer = styled.View`
+  flex-direction: row;
+  justify-content: space-between;
+  width: 80%;
 `;
 
-export const ImageContainer = styled.View`
-  flex-direction: row;
+export const ProductInfoImageContainer = styled.View`
+  margin-top: 7.5%;
```
Comment (emilyzhong) [inline styled/product.js]: Similarly with margin (+ @ line 25) -- pixels

File: styled/product.js [hunk 12]
```diff
@@ -39,5 +42,9 @@ export const styles = StyleSheet.create({
   storesModal: {
     backgroundColor: 'white',
     opacity: 0.8
+  },
+  horizontalScroll: {
+    padding: '1%',
```
Comment (emilyzhong) [inline styled/product.js]: px

File: styled/store.js [hunk 13]
```diff
@@ -9,22 +12,77 @@ export const StoreModalBar = styled.View`
...
+  padding: 4%;
 `;
 
-export const SearchBar = styled.TouchableOpacity`
-  background-color: white;
-  align-items: center;
-  margin: 1%;
-  border-radius: 5px;
+export const StoreListHeaderContainer = styled.View`
+  background-color: #f07723;
```
Comment (emilyzhong) [inline styled/store.js]: `Colors.primaryOrange`

File: styled/store.js [hunk 14]
```diff
@@ -9,22 +12,77 @@ export const StoreModalBar = styled.View`
...
+  opacity: 0.8;
+`;
+
+export const StoreCardContainer = styled.View`
+  margin: 7.5px;
```
Comment (emilyzhong) [inline styled/store.js]: `margin: 8px` (?) I would check the figma file to be sure

File: styled/store.js [hunk 15]
```diff
@@ -9,22 +12,77 @@ export const StoreModalBar = styled.View`
...
 
-export const TopText = styled.Text`
-  font-style: normal;
-  font-weight: 900;
+export const StoreListTitle = styled(Title)`
   text-align: center;
-  margin-top: 10px;
-  font-size: 25px;
+  color: white;
+  margin-top: 2.5%;
```
Comment (emilyzhong) [inline styled/store.js]: px margins!

File: styled/store.js [hunk 16]
```diff
@@ -9,22 +12,77 @@ export const StoreModalBar = styled.View`
...
+`;
+
+export const SearchBar = styled(ButtonContainer)`
+  display: flex;
+  flex-direction: row
+  align-items: center;
+  justify-content: center;
+  margin: 20px 25%;
+  height: 5%
```
Comment (emilyzhong) [inline styled/store.js]: also pixel heights

File: styled/store.js [hunk 17]
```diff
@@ -9,22 +12,77 @@ export const StoreModalBar = styled.View`
...
+`;
+
+// FlatList styles
+
+export const styles = StyleSheet.create({
+  inputContainer: {
+    backgroundColor: 'white'
+  },
+  container: {
+    backgroundColor: '#f07723',
```
Comment (emilyzhong) [inline styled/store.js]: i think this is primary orange pls use color variables!

File: styled/store.js [hunk 18]
```diff
@@ -9,22 +12,77 @@ export const StoreModalBar = styled.View`
...
+    borderBottomColor: 'transparent',
+    borderTopColor: 'transparent'
+  },
+  input: {
+    fontFamily: 'Arial' // @tommpoa TODO – load poppins!
+  },
+  searchIcon: {
+    color: '#f07723'
```
Comment (emilyzhong) [inline styled/store.js]: color

File: screens/map/ProductsScreen.js [hunk 19]
```diff
@@ -1,40 +1,52 @@
...
+    return products.map((product, i) => {
+      return (
+        <ProductCard
+          key={i}
+          product={product}
+          navigation={this.props.navigation}
+          store={store}
+        />
+      );
+    });
```
Comment (wangannie) [inline screens/map/ProductsScreen.js]: This fix was suggested by Myles to maintain consistency.

File: styled/store.js [hunk 20]
```diff
@@ -89,9 +91,6 @@ export const styles = StyleSheet.create({
     borderTopColor: 'transparent'
   },
   input: {
-    fontFamily: 'Arial' // @tommpoa TODO – load poppins!
-  },
-  searchIcon: {
-    color: Colors.primaryOrange
+    fontFamily: 'poppins-regular' // @tommpoa TODO – load poppins!
```
Comment (wangannie) [inline styled/store.js]: Oops remove this TODO

File: styled/shared.js [hunk 21]
```diff
@@ -1,36 +1,33 @@
...
 
-export const ButtonText = styled.Text`
-  color: white;
-  padding: 7.5px 7.5px;
+export const ScrollCategory = styled.Text`
+  font-size: 14px;
+  font-weight: normal;
 `;
 
 export const PoppinsText = styled.Text`
```
[the remark this answers is not in the document]
Comment (wangannie) [reply styled/shared.js]: It's still being used by News atm. We'll remove it once News is styled with the base components.

**Comment by wangannie** [comment]
## What's New (pt 2)
### Styling Tweaks
- Updated language across the app to match Figma (ex. 'Browsing healthy products at')
- Updated alignment and spacing of the EBTStatusBar to precisely match designs
- Created `DividerBar` component for StoreCard to display in the store list
- Fixed the tiny slit between `BottomSheetHeaderContainer` and `BottomSheetContainer`
- Fixed inconsistent alignment on the bottom sheet
- Fixed spacing between store cards on the Store List screen
- Fixed spacing and alignment on store details between icons and text
- Fixed alignment issues with the icon and text on the map view search bar
- Fixed the react-native SearchBar styling to use Poppins and replaced the default icon with FontAwesome
- Added drop shadows to MapScreen search bar and header containers on store list and products list screens
- Added drop shadow to the bottom sheet — solved early issues with the shadow leaking around the edges
- Added and styled the header bar in Product List with `ProductListHeaderContainer` to match the designs and for consistency with StoreList

### Cleaning
- Standardized icons to use FontAwesome5 and fixed icon sizing
- Converted product list from FlatList to a `renderProductList` function to maintain consistency and make it easier to style
- Removed deprecated filtered renderProducts function
- Replaced non-white hex codes with standard colors from `Colors.js`

### Interactions
- Updated the bottom sheet to default to the expanded products view upon tapping 'SEE PRODUCTS' from the store list
- Set focus to the search bar upon clicking search from MapScreen
- Fixed messy BottomSheet behavior so it can no longer be fully uprooted, reduced bouncing behavior
Stores:Products Interactions

### Misc.
- Updated Airtable call in `mapUtils` to see if a store accepts rewards to display an additional detail line to indicate that rewards are accepted

### Screenshots

## Bugs/Todos
- Update StoreCards to indicate where rewards are not accepted
- Map styling - location pins, recenter location button
- Fix 'See Products' button alignment and handle store title overflow
- Sometimes 'Find a Store' on the map screen search bar floats off but usually goes back to normal upon refreshing

Untitled 2
- Nit: Figure out a way to handle the bottom margin of the store list header container so store cards don't go under this white bar. Same for the left and right margins on the products list in the map screen.
Untitled
Untitled 2

File: components/product/ProductInfo.js [hunk 22]
```diff
@@ -10,15 +13,19 @@ import SpecialNotice from './SpecialNotice';
...
-      </View>
-      <SpecialNotice />
-    </View>
+      <ProductInfoCaptionContainer>
+        <Caption>${customerCost} each</Caption>
+        <Caption>Points Earned: {points}</Caption>
+      </ProductInfoCaptionContainer>
+      <ProductNoticeContainer>
+        <Body>
+          Note: not all products listed are available at every store. Please
```
Comment (thumn) [inline components/product/ProductInfo.js]: smol thing: capitalize the "n" in "not"? :0

Comment (thumn) [reply components/product/ProductInfo.js]: I couldn't find the exact prototype for this in figma but correct me if I'm wrong

File: components/product/ProductInfo.js [hunk 23]
```diff
@@ -10,15 +13,19 @@ import SpecialNotice from './SpecialNotice';
...
       <Title>{name}</Title>
-      <Subtitle>Category: {category}</Subtitle>
-      <View style={{ flexDirection: 'row' }}>
-        <Subtitle>${customerCost} each</Subtitle>
-        <Subtitle>Points Earned: {points}</Subtitle>
-      </View>
-      <SpecialNotice />
-    </View>
+      <ProductInfoCaptionContainer>
+        <Caption>${customerCost} each</Caption>
```
Comment (thumn) [inline components/product/ProductInfo.js]: Should probably be rounded to the nearest two decimal places (i.e. for banana it's showing up as $0.5 right now)

Comment (anniero98) [reply components/product/ProductInfo.js]: +1, try `toFixed(2)` !

File: components/product/ProductInfo.js (hunk 22 above)
Comment (anniero98) [reply components/product/ProductInfo.js]: iirc it's in the Figma from last semester, was mistakenly removed but should be added back soon :o !

File: components/store/StoreCard.js [hunk 24]
```diff
@@ -38,6 +38,20 @@ function StoreCard({ store, callBack, seeProduct }) {
...
+      {rewards && (
+        <InLineContainer style={{ alignItems: 'center' }}>
+          <FontAwesome5
+            name="star"
+            solid
+            size={16}
+            color={Colors.primaryGreen}
+          />
+          <Body color={Colors.primaryGreen}>
+            {' '}
```
Comment (tommypoa) [inline components/store/StoreCard.js]: Line 50: probably can remove?

File: screens/map/ProductsScreen.js [hunk 25]
```diff
@@ -1,40 +1,52 @@
...
-import { FlatList } from 'react-native';
+import { FlatList, View } from 'react-native';
 import { ScrollView } from 'react-native-gesture-handler';
 import ProductCard from '../../components/product/ProductCard';
-import { styles } from '../../styled/product';
+import {
+  ProductListHeaderContainer,
+  ProductListContainer,
+  ProductListTitle,
+  styles
```
Comment (tommypoa) [inline screens/map/ProductsScreen.js]: Now that `FlatList` is removed, we don't need to import `styles` from `/styled/product`

**Comment by tommypoa** [comment]
Read through all the changes - made some small fixes according to all reviewer comments as well. Awesome stuff @wangannie - feel free to merge after you've looked through the changes I made.

**Pull request merged**